### PR TITLE
feat(release): group standing-PR selection checklist by channel

### PR DESCRIPTION
**Pull request by goosewobbler**
## What

The standing-PR **Packages to release** checklist now groups rows by release channel: a **Stable** section (advancing on `latest`) and a **Prereleases** section, with each prerelease row surfacing the dist-tag channel it advances on (e.g. `→ 2.0.0-next.1 · next`). Closes #487.

Before, per-package channels (#485) meant a standing PR could mix stable and prerelease packages in one flat list, so maturity wasn't visible at a glance.

## How

- Rows are bucketed by `update.channel ?? deriveReleaseChannel(newVersion)` (the second clause keeps old manifests working).
- **Flat** render partitions updates per channel and reuses the existing flat section renderer unchanged.
- **Hierarchical** render computes the hierarchy once, then buckets whole units by their primary's channel (single-channel by construction) and orphans by their own — so `computeHierarchy`, `primaryRow`, `childBullet`, and the streamlined/granular logic are preserved within each section.
- Section headings appear only when more than one channel is present; stable-only PRs render byte-for-byte as before.
- The per-row changelog renderer (#488) is reused unchanged — this PR changes *where* rows are placed, not *how* changelogs attach.

## Scope

Purely presentational — selection/hold-back semantics, marker round-tripping, and version computation are untouched. Leaves a marked seam in `versionDisplay` for the per-package "graduate to stable" affordance (#486), which renders nothing yet.

## Tests

New `channel-grouped sections (#487)` suite covers mixed / stable-only / prerelease-only, per-row version + dist-tag, the old-manifest channel-derivation fallback, deselection round-trip across sections, and two hierarchical cases. Full gate green (`lint typecheck test`, 722 release tests pass).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<details open><summary><h3>Greptile Summary</h3></summary>

This PR groups the standing-PR **Packages to release** checklist by release channel, partitioning updates into a **Stable** section and a **Prereleases** section; single-channel PRs continue to render an unheaded flat list (byte-for-byte backward compatible). Each prerelease row now appends the npm dist-tag derived via `getDistTag` (e.g. `→ 2.0.0-next.1 · \`next\``).

- `byChannel` / `channelOf` / `unitChannel` bucket flat updates and hierarchical units into per-channel groups; `renderFlat` and `renderHierarchical` iterate `CHANNEL_ORDER` and call `pushSectionHeading` only when more than one channel is present.
- `versionDisplay` is a new shared label formatter that delegates to `getDistTag` for prerelease rows; `row`, `primaryRow`, and `childBullet` all use it, so the dist-tag appears consistently in every render path.
- The known caveat around cross-channel prerequisite nesting (the `↳` label is lost when target and prerequisite land in different channel sections) is now documented in the `renderFlat` JSDoc, and the corresponding `selectionWarnings` guard makes the scenario semantically invalid in practice.
</details>

<h3>Confidence Score: 5/5</h3>

Purely presentational change — selection/hold-back semantics, marker round-tripping, and version computation are untouched. Single-channel PRs render byte-for-byte as before.

The bucketing logic is exhaustive over a two-value `ReleaseChannel` type, the `versionDisplay` helper is consistently applied across all three row renderers, and the deselection round-trip test confirms markers survive the new dist-tag suffix. The known cross-channel prerequisite nesting edge case is properly documented and guarded upstream by `selectionWarnings`.

No files require special attention.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/release/src/standing-pr/selection-region.ts | Core change: adds channel-bucketing helpers (`byChannel`, `channelOf`, `unitChannel`), refactors flat and hierarchical renderers to partition into Stable/Prereleases sections, and updates `versionDisplay` to append the npm dist-tag for prerelease rows. Logic is sound; the caveat about cross-channel prerequisites is now documented in the `renderFlat` JSDoc. |
| packages/release/test/unit/selection-region.spec.ts | New `channel-grouped sections (#487)` suite adds 7 cases: mixed/stable-only/prerelease-only flat, per-row dist-tag rendering, old-manifest fallback, deselection round-trip, and two hierarchical unit scenarios. Coverage is thorough and tests map directly to the invariants described in the PR. |
| packages/release/docs/ci-setup.md | One-sentence addition to the ad-hoc selection paragraph explaining the Stable/Prereleases split and the single-channel flat-list fallback. Accurate and matches the implementation. |

</details>

<details><summary><h3>Flowchart</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[renderSelectionRegion] --> B{primary config?}
    B -- yes --> C[renderHierarchical]
    B -- no --> D[renderFlat]

    C --> E[computeHierarchy]
    E --> F[byChannel units via unitChannel]
    E --> G[byChannel orphans via channelOf]
    F & G --> H{channels.length > 1?}
    H -- yes --> I[pushSectionHeading per channel]
    H -- no --> J[no headings]
    I & J --> K[renderUnit per unit]
    I & J --> L[renderOrphan per orphan]

    D --> M[byChannel updates via channelOf]
    M --> N{channels.length > 1?}
    N -- yes --> O[pushSectionHeading per channel]
    N -- no --> P[no headings]
    O & P --> Q[renderFlatSection per channel]

    K & L & Q --> R[versionDisplay]
    R --> S{channelOf = prerelease?}
    S -- yes --> T["version · distTag (getDistTag)"]
    S -- no --> U[version only]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A[renderSelectionRegion] --> B{primary config?}
    B -- yes --> C[renderHierarchical]
    B -- no --> D[renderFlat]

    C --> E[computeHierarchy]
    E --> F[byChannel units via unitChannel]
    E --> G[byChannel orphans via channelOf]
    F & G --> H{channels.length > 1?}
    H -- yes --> I[pushSectionHeading per channel]
    H -- no --> J[no headings]
    I & J --> K[renderUnit per unit]
    I & J --> L[renderOrphan per orphan]

    D --> M[byChannel updates via channelOf]
    M --> N{channels.length > 1?}
    N -- yes --> O[pushSectionHeading per channel]
    N -- no --> P[no headings]
    O & P --> Q[renderFlatSection per channel]

    K & L & Q --> R[versionDisplay]
    R --> S{channelOf = prerelease?}
    S -- yes --> T["version · distTag (getDistTag)"]
    S -- no --> U[version only]
```

</a>
</details>

<sub>Reviews (2): Last reviewed commit: ["docs(release): note cross-channel prereq..."](https://github.com/goosewobbler/releasekit/commit/832aebc528e3bfb110d5c32bf7d37316fcd1550a) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=40667286)</sub>

### CODE REVIEW

File: packages/release/src/standing-pr/selection-region.ts [hunk 1]
```diff
@@ -243,6 +301,71 @@ function renderFlat(
   }
 }
```
[the remark this answers is not in the document]
Comment (goosewobbler) [reply packages/release/src/standing-pr/selection-region.ts:302]: Documented in 832aebc5 (caveat on `renderFlat`). One correction on the guard: `selectionWarnings` only warns when a prerequisite is *held back* (deselected), not on a channel mismatch — so a *selected* `-next` prerequisite of a stable target isn't actually warned. It's a plausible mixed-maturity config (exactly what #485 enables), but the degradation is purely cosmetic (loses the `↳ prerequisite` label) and only in flat mode; the hierarchical/streamlined path keeps units intact. Left flat-mode rendering as-is and documented the channel-grouping-wins-over-nesting behaviour rather than reworking it for this rare edge.
